src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\QueryHandler;
+
+use Customer;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Query\SearchCustomers;
+use PrestaShop\PrestaShop\Core\Domain\Customer\QueryHandler\SearchCustomersHandlerInterface;
+
+/**
+ * Handles command that searches for customers
+ *
+ * @internal
+ */
+final class SearchCustomersHandler implements SearchCustomersHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SearchCustomers $query)
+    {
+        $queries = explode(' ', $query->getQuery());
```

Comment:
legacy query is just a `string` (e.g. "John Doe", "John" & etc.), but internally it splits this query into multiple parts as you can see. is this really a problem? 

> Legacy query must be transformed into something more usable

what do you mean? :thinking:

Reply:
These multiple parts are structured data. Instead of carrying the string as it was provided `John Doe Maria John` it should be an array `$searchedWords = ['John', 'Doe', 'Maria', 'John'];`

Reply:
ok, we can have `searchWords` or similar as command attribute, but `Customer::searchByName($word)` searches it between 'id_customer', 'firstname', 'lastname', 'email' fields in db. so it doesnt matter whether its name or id, it will perform search on all columns. 

we could not relay on legacy at all and perform custom sql query from handler.

Reply:
😝 so legacy `Customer::searchByName()` is actually a `Customer::searchByIdAndFirstnameAndLastnameAndEmail()` ...

Reply:
If you look at `Customer::searchByName()` logic, it actually is a custom SQL query. SO let's keep using it. The only thing we could improve here is the name of the function, but that would be a BC break or an alias, so not today ⚔️ this is out of migration scope.